lib/workshops/src/fetcher_impl.dart

```diff
@@ -12,29 +12,52 @@ abstract class WorkshopFetcherImpl implements WorkshopFetcher {
   Future<Workshop> fetch() async {
     var metadata = await fetchMeta();
     var steps = await fetchSteps(metadata);
-    return Workshop(metadata.name, metadata.type, steps);
+    return Workshop(metadata.name, metadata.type, steps.toList());
   }
 
   Future<Meta> fetchMeta() async {
     var contents = await loadFileContents(['meta.yaml']);
     return checkedYamlDecode(contents, (Map m) => Meta.fromJson(m));
   }
 
-  Future<List<Step>> fetchSteps(Meta metadata) async {
-    var steps = <Step>[];
-    for (var stepConfig in metadata.steps) {
-      steps.add(await fetchStep(stepConfig));
+  Future<Iterable<Step>> fetchSteps(Meta metadata) async {
+    // The unnamed list constructor was removed in Dart 2.12, so use a map
+    // instead of a list to fetch each step in parallel and place the results in
+    // the original order.
+    var map = <int, Step>{};
+    var futures = <Future>[];
+    for (var i = 0; i < metadata.steps.length; i++) {
+      var config = metadata.steps[i];
+      var future = fetchStep(config).then((step) => map[i] = step);
+      futures.add(future);
     }
-    return steps;
+    await Future.wait(futures);
```

Comment:
Just curious: could the return value of `Future.wait` be used as the function's return value, sparing you from having to make an intermediate map, or would that approach run into the problem you reference in the comment above?